crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -411,3 +473,185 @@ impl<'a> ReflectEnum<'a> {
         &self.variants
     }
 }
+
+/// Represents a path to a type.
+pub(crate) enum PathToType<'a> {
+    /// Types without a crate/module that can be named from any scope (e.g. `bool`).
+    Primitive(&'a Ident),
+    /// Using `::my_crate::foo::Bar` syntax.
+    ///
+    /// May have a seperate alias path used for the `TypePath` implementation.
+    External { path: &'a Path, alias: Option<Path> },
+    /// The name of a type relative to its scope.
+    /// The type must be able to be named from just its name.
+    ///
+    /// May have a seperate alias path used for the `TypePath` implementation.
+    ///
+    /// Module and crate are found with [`module_path!()`](core::module_path),
+    /// if there is no alias specified.
+    Internal {
+        name: &'a Ident,
+        alias: Option<Path>,
+    },
+    /// Any [`syn::Type`] with only a defined `type_path` and `short_type_path`.
+    #[allow(dead_code)]
+    // Not currently used but may be useful in the future due to its generality.
+    Anonymous {
+        qualified_type: Type,
+        long_type_path: proc_macro2::TokenStream,
+        short_type_path: proc_macro2::TokenStream,
+    },
+}
+
+impl<'a> PathToType<'a> {
+    /// Returns the path interpreted as an [`struct@Ident`],
+    /// or [`None`] if anonymous or primitive.
+    fn named_as_ident(&self) -> Option<&Ident> {
```

Comment:
A lot of these methods are prefixed with `named_`. Are these meant to read `name`? Otherwise, they all sound like they should return `bool` ("is this named as an ident?").